### PR TITLE
CEDS-2106 refactoring and preparation

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: test/unit/uk/gov/hmrc/exports/movements/base/CustomsExportsBaseSpec.scala [hunk 1]
```diff
@@ -76,7 +76,7 @@ trait CustomsExportsBaseSpec
 
   def messagesApi: MessagesApi = injector.instanceOf[MessagesApi]
 
-  def injector: Injector = app.injector
+  def injector: inject.Injector = app.injector
```
Comment (PatrykRudnicki) [inline test/unit/uk/gov/hmrc/exports/movements/base/CustomsExportsBaseSpec.scala]: Why this change? Still it should recognise `Injector`

Comment (bambuchaAdm) [reply test/unit/uk/gov/hmrc/exports/movements/base/CustomsExportsBaseSpec.scala]: name clash with test util probably

Comment (bambuchaAdm) [reply test/unit/uk/gov/hmrc/exports/movements/base/CustomsExportsBaseSpec.scala]: We found this trait is not used anywhere - we would remove it

File: test/unit/uk/gov/hmrc/exports/movements/base/Injector.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  /**
+    * Clearing shared metrics registries to avoid `A metric named jvm.attribute.vendor already exists` error.
+    *
+    * It appears very often with places with injector. This is enough solution for this problem.
+    *
+    * Reference and other solutions: https://github.com/kenshoo/metrics-play/issues/74
+    */
+  SharedMetricRegistries.clear()
+
+  private val injector = GuiceApplicationBuilder().injector()
```
Comment (bambuchaAdm) [inline test/unit/uk/gov/hmrc/exports/movements/base/Injector.scala]: Let's move clear declaration to varaible initialization - i have always problem with those that its order is not really known
```
private val injector =  {
  SharedMetricRegistries.clear()
  GuiceApplicationBuilder().injector()
}
```
